lib/sdk/writer/index.js

```diff
@@ -48,29 +102,200 @@ class ImageWriter extends EventEmitter {
     this.pipeline = null
     this.target = null
 
-    this.hadError = false
-
     this.bytesRead = 0
     this.bytesWritten = 0
     this.checksum = {}
   }
 
+  /**
+   * @summary Verify that the selected destination device exists
+   * @param {Function} callback - callback(error)
+   * @private
+   * @example
+   * writer.checkSelectedDevice((error) => {
+   *   // ...
+   * })
+   */
+  checkSelectedDevice (callback) {
+    debug('state:device-select', this.options.path)
+    this.destinationDevice = null
+    drivelist.list((error, drives) => {
+      debug('state:device-select', this.options.path, error ? 'NOT OK' : 'OK')
+
+      if (error) {
+        callback.call(this, error)
+        return
+      }
+
+      const selectedDrive = _.find(drives, {
+        device: this.options.path
+      })
+
+      if (!selectedDrive) {
+        const selectionError = errors.createUserError({
+          title: 'The selected drive was not found',
+          description: `We can't find ${this.options.path} in your system. Did you unplug the drive?`,
+          code: 'EUNPLUGGED'
+        })
+        debug('state:device-select', this.options.path, 'NOT OK')
+        callback.call(this, selectionError)
+        return
+      }
+
+      this.destinationDevice = selectedDrive
+
+      callback.call(this)
+    })
+  }
+
+  /**
+   * @summary Unmount the destination device
+   * @param {Function} callback - callback(error)
+   * @private
+   * @example
+   * writer.unmountDevice((error) => {
+   *   // ...
+   * })
+   */
+  unmountDevice (callback) {
+    if (os.platform() === 'win32') {
```

Comment:
Would have been nice if the comment at https://github.com/resin-io/etcher/pull/2080/files#diff-e662af03d4832b4cbfad654fb2c58554L80 could have been preserved ;-)